fdbserver/OldTLogServer_6_0.actor.cpp

```diff
@@ -695,9 +695,14 @@ ACTOR Future<Void> tLogPopCore( TLogData* self, Tag inputTag, Version to, Refere
 	}
 	state Version upTo = to;
 	int8_t tagLocality = inputTag.locality;
-	if (logData->logSystem->get().isValid() && logData->logSystem->get()->isPseudoLocality(tagLocality)) {
-		upTo = logData->logSystem->get()->popPseudoLocalityTag(tagLocality, to);
-		tagLocality = tagLocalityLogRouter;
+	if (isPseudoLocality(tagLocality)) {
+		if (logData->logSystem->get().isValid()) {
+			upTo = logData->logSystem->get()->popPseudoLocalityTag(inputTag, to);
+			tagLocality = tagLocalityLogRouter;
+		} else {
+			TraceEvent("TLogPopNoLogSystem", self->dbgid).detail("Locality", tagLocality).detail("Version", upTo);
```

Comment:
should this be SevError?

Reply:
For the record, the backup worker will continue pop tags after this version, so in normal operations this is safe. If there is a recovery and the last pop was lost, then the logic in `commitMessages()` will index this tag to `tag.id = tag.id % logData->logRouterTags;`, which saves us if new epoch has less log router tags. So we are safe to ignore the pop in the `else` case.